### PR TITLE
Add testing of example scripts (#1041)

### PR DESCRIPTION
**Pull request by eleftherioszisis**
Cherry-picked from v4 branch

### CODE REVIEW

File: tests/test_examples.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    spec = importlib.util.spec_from_file_location(filepath.stem, filepath)
+    module = spec.loader.load_module()
+
+    with tempfile.TemporaryDirectory() as tempdir:
+
+        # change directory to avoid creating files in the root folder
+        try:
+            cwd = os.getcwd()
+            os.chdir(tempdir)
+            module.main()
```
Comment (adrien-berchet) [inline tests/test_examples.py:25]: Isn't it better to call a subprocess here to be sure that it works in a clean python engine?

Comment (adrien-berchet) [reply tests/test_examples.py:25]: Forget about this :smile:

Comment (eleftherioszisis) [reply tests/test_examples.py:25]: Nah, I prefer it this way. It is also easier to debug when it fails. It is highly unlikely it won't work in a subprocess if it works like this.

**Review by adrien-berchet — APPROVED** [review]

File: tests/test_examples.py (hunk 1 above)
Comment (adrien-berchet) [reply tests/test_examples.py:25]: Yeah I agree :)

**Pull request merged**